### DISCUSSION TITLE
Equality change in v7.13.0 for getValues

### DISCUSSION THREAD

**Discussion by joniaiuser** [react-hook-form/react-hook-form#6315] [Q&A]
I have a form field that is an array of objects.

[{field1,
field2,
field3,}]

I notice that when I modify the field1, field2 values after registering them, i.e.
```
register(`formData.${idx}.field1` as "formData.0.field1")
```
 even though the nested values from getValues() do change when i console log them, the old getValues()  === new getValues().

This is causing a bug for my app where the useEffect no longer triggers an update. Is this change in behaviour expected/intended and what should be the workaround?

**Comment by joniaiuser** [comment]
Ok an obvious workaround is to include the nested values in the useEffect and useMemo dependency arrays, however this change would span across the entire codebase. Is this the only approach?

**Reply by bluebill1049** [reply]
what's your usage like? and why getValues() results is used as deps?
